### PR TITLE
Add a direct test for the "other attendees on these days" query

### PR DESCRIPTION
**Pull request by stefan-burke**
## What this does

Our unit-tests report flags source files that don't yet have their own dedicated test. It pointed at `src/shared/db/attendees/overlap.ts` — the query behind the Logistics tab's "Other Attendees" list, which shows everyone else booked on the same days as the attendee you're looking at.

That query had no test of its own. It was only ever run indirectly, as a side effect of a bigger test that loads the whole Logistics page. So if someone changed the query and got the dates slightly wrong, nothing would have clearly caught it.

This adds a small, focused test file (`test/shared/db/attendees/overlap.test.ts`) that checks the query directly.

## What it checks

- Two bookings that only just touch — one ending on the exact day the other begins — are **not** counted as overlapping. Days run right up to but not including the end date.
- A booking that sits entirely outside the days you're looking at is left out.
- The attendee you're currently viewing never appears in their own "other attendees" list.
- Bookings with no quantity, and bookings with no dates at all, are left out.
- The results come back earliest-first, and each one carries the right listing, quantity, and dates.

## Why

No behaviour changes — this is test-only. It locks in how the "who else is here on these days" list decides who counts, so a future edit to that query can't quietly break it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_011GrobXJsoo8StWaWFbwFo1)_

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive coverage for detecting overlapping bookings.
  * Validates boundary conditions, excluded records, zero-quantity bookings, date-less bookings, result ordering, and returned booking details.

### CODE REVIEW

File: test/shared/db/attendees/overlap.test.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import { describeWithEnv } from "#test-utils/db.ts";
+import {
+  createDailyTestListing,
+  createTestListing,
+} from "#test-utils/db-helpers/listings.ts";
+
+/** The window every test queries against: 2030-03-10 (inclusive) to 2030-03-12
+ *  (exclusive), built the same way the booked ranges are so the boundary
+ *  comparisons use matching stored formats. */
+const WINDOW = dateToRange("2030-03-10", 2);
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/shared/db/attendees/overlap.test.ts:25]: You're right, and this was the important one. Fixed in follow-up PR #1759 (commit 373756c): the window bounds are now anchored byte-for-byte to the adjacent rows' stored ranges (`startAt === adjacentBefore.end_at`, `endAt === adjacentAfter.start_at`), so at each boundary the row's instant equals the window's exactly and only the strictness of `<`/`>` decides inclusion — not the `Z` vs `.000Z` suffix. Verified by temporarily switching the query to `<=`/`>=` and confirming both boundary tests then fail.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/shared/db/attendees/overlap.test.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      ids: results.map((row) => row.attendee_id),
+      inside,
+      results,
+      self,
+    };
+  };
+
+  test("lists other attendees' overlapping bookings, earliest start first", async () => {
+    const f = await overlapFixture();
+    expect(f.ids).toEqual([f.before, f.inside]);
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/shared/db/attendees/overlap.test.ts:94]: Fixed in follow-up PR #1759 (commit 373756c): the fixture now creates the later-dated overlapper (`inside`) before the earlier-dated one (`before`), so `before` has the higher id but the earlier date. `expect(f.ids).toEqual([f.before, f.inside])` then only holds under a `start_at` sort and fails if the query orders by attendee id.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/shared/db/attendees/overlap.test.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+const book = async (
+  name: string,
+  listingId: number,
+  booking: { date?: string; durationDays?: number; quantity?: number } = {},
+): Promise<number> => {
+  const result = await createAttendeeAtomic({
+    bookings: [{ listingId, ...booking }],
+    email: "",
+    name,
+  });
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/shared/db/attendees/overlap.test.ts:38]: Added in follow-up PR #1759 (commit 373756c): a test books an overlapping row, flips it to a servicing hold with `UPDATE attendees SET kind = ?` using the `SERVICING_KIND` constant, and asserts it's excluded — so removing the `attendee.kind = 'attendee'` filter now fails a test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/shared/db/attendees/overlap.test.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  // The fixtures always have capacity, so the create always succeeds.
+  return (result as Extract<typeof result, { success: true }>).attendees[0]!.id;
+};
+
+describeWithEnv("db > attendees > overlap", { db: true }, () => {
+  /** A daily listing with a cast of bookings around the query window: one for
+   *  the "current" attendee (excluded), two genuine overlappers, and three that
+   *  must be filtered out (two exactly adjacent, one far away). */
+  const overlapFixture = async () => {
+    const daily = await createDailyTestListing({ maxAttendees: 50 });
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/shared/db/attendees/overlap.test.ts:48]: Added in follow-up PR #1759 (commit 373756c): a test books the excluded "current" attendee on listing A and another attendee on a second daily listing B in the same window, then asserts the listing-B attendee is still returned — so narrowing the query to the current attendee's listing(s) would fail it.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/shared/db/attendees/overlap.test.ts [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    });
+    const before = await book("Before Overlap", daily.id, {
+      date: "2030-03-08",
+      durationDays: 3,
+    });
+    const inside = await book("Inside", daily.id, {
+      date: "2030-03-11",
+      durationDays: 1,
+      quantity: 2,
+    });
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/shared/db/attendees/overlap.test.ts:61]: Added in follow-up PR #1759 (commit 373756c): a test books one attendee with two overlapping booking lines (on two listings) and asserts both rows come back, keyed by listing_id — so a regression that DISTINCTs/groups by attendee and drops extra rows now fails.

---
_Generated by [Claude Code](https://claude.ai/code)_
